app/controllers/books_controller.rb

```diff
@@ -1,5 +1,6 @@
 class BooksController < ApplicationController
-    before_action :set_book, only: %i[show edit update]
+    before_action :set_book, only: %i[show edit update reserve]
+    before_action :authenticate_user!
```

Comment:
**[Minor]** I'm not 100% sure about this one, but you can probably check this with a debugger call in both methods: you usually want this `before_action :authenticate_user!` to be declared _before_ `before_action :set_book` so that the user is redirected right away if there's no session. 😄 

I believe they'll run in the order they're declared otherwise, so if your `before_action` relied on the `current_user` you could get an exception or unexpected behavior there (e.g., doing something like `current_user.reservations.find(...)`).